### PR TITLE
Add `Flame::Router::Routes#to_s`, for routes printing

### PR DESCRIPTION
**Pull request by AlexWayfer**
```ruby
puts MyApplication.router.routes
```

### CODE REVIEW

**Comment by AlexWayfer** [comment]
@JelF, what do you think?

File: lib/flame/router/routes.rb [hunk 1]
```diff
@@ -70,6 +70,29 @@ def allow
...
+						path = Flame::Path.new(prefix, key)
+						result << value.to_s(path)
+					end
+				end
+			end
+
+			## Sort routes for human readability
+			def sort
+				sort_by { |key, _value| key.is_a?(Symbol) ? 1 : 2 }
+			end
```
[the remark this answers is not in the document]
Comment (AlexWayfer) [reply lib/flame/router/routes.rb]: I think I made all these things.

File: spec/unit/router/routes_spec.rb [hunk 2]
```diff
@@ -162,4 +162,22 @@ def deep_check(values)
 			it { is_expected.to be_nil }
 		end
 	end
+
+	describe '#to_s' do
+		subject { super().to_s }
```
[the remark this answers is not in the document]
Comment (AlexWayfer) [reply spec/unit/router/routes_spec.rb]: OK, done.

File: spec/unit/router/routes_spec.rb [hunk 3]
```diff
@@ -162,4 +162,22 @@ def deep_check(values)
...
+		end
+
+		it do
+			is_expected.to eq <<~OUTPUT
+				\e[1m   GET /foo/bar\e[22m
+				       \e[3m\e[36m42\e[0m\e[23m
+				\e[1m  POST /foo/bar/baz\e[22m
+				       \e[3m\e[36m84\e[0m\e[23m
+			OUTPUT
+		end
```
[the remark this answers is not in the document]
Comment (AlexWayfer) [reply spec/unit/router/routes_spec.rb]: OK, done.

File: lib/flame/router/routes.rb [hunk 4]
```diff
@@ -70,6 +70,29 @@ def allow
...
+						result << <<~OUTPUT
+							\e[1m#{format PADDING_FORMAT, key} #{prefix}\e[22m
+							#{' ' * PADDING_SIZE} \e[3m\e[36m#{value}\e[0m\e[23m
+						OUTPUT
+					else
+						path = Flame::Path.new(prefix, key)
+						result << value.to_s(path)
+					end
+				end
+			end
```
[the remark this answers is not in the document]
Comment (AlexWayfer) [reply lib/flame/router/routes.rb]: `each_with_object` is `1.03`–`1.05` slower and greedy than `map` + `join`. Thank you, replaced.

Anyway, it's not critical at application start, I think.

[the remark this answers is not in the document]
Comment (JelF) [reply lib/flame/router/routes.rb]: > Anyway, it's not critical at application start, I think.

I mentioned `map {}.join` as most verbose :)

File: spec/unit/router/routes_spec.rb [hunk 5]
```diff
@@ -162,4 +162,46 @@ def deep_check(values)
...
 	end
+
+	describe '#to_s' do
+		subject { routes.to_s }
+
+		before do
+			routes['foo']['bar'][:GET] = 42
+			routes['foo']['bar']['bar'] = described_class.new
+			routes['foo']['bar']['bar'][:POST] = 36
+			routes['foo']['bar']['baz'][:DELETE] = 84
```
Comment (JelF) [inline spec/unit/router/routes_spec.rb]: If one 'baz' will be above 'bar' here, it will increase verbosity of sorting

Comment (AlexWayfer) [reply spec/unit/router/routes_spec.rb]: Done, thank you.

**Pull request merged**